v03_pipeline/lib/tasks/reference_data/base_updated_reference_dataset.py

```diff
@@ -0,0 +1,52 @@
+import hail as hl
+import luigi
+
+from luigi_pipeline.lib.hail_tasks import GCSorLocalTarget
+from v03_pipeline.lib.paths import valid_reference_dataset_path
+from v03_pipeline.lib.reference_data.reference_dataset import ReferenceDataset
+from v03_pipeline.lib.tasks.base.base_loading_run_params import BaseLoadingRunParams
+from v03_pipeline.lib.tasks.base.base_update import BaseUpdateTask
+
+
+@luigi.util.inherits(BaseLoadingRunParams)
+class UpdatedReferenceDataset(BaseUpdateTask):
+    reference_dataset: ReferenceDataset
+
+    def output(self):
+        return GCSorLocalTarget(
+            valid_reference_dataset_path(
+                self.reference_genome,
+                self.reference_dataset,
+            ),
+        )
+
+    def initialize_table(self):
+        key_type = self.reference_dataset.table_key_type(
+            self.reference_genome,
+        )
+        return hl.Table.parallelize(
+            [],
+            key_type,
+            key=key_type.fields,
+            globals=hl.Struct(
+                version=hl.missing(hl.tstr),
+                enums=hl.Struct(),
+            ),
+        )
+
+    def update_table(self, ht):
+        ht_version = hl.eval(ht.globals).version
```

Comment:
I'm using `BaseUpdateTask` but technically this flow can be different in that if the table doesn't exist, we create it. If the table exists but the version (and we don't even have to read the table we can just read the name) is different, also write it. We're no longer updating an existing table. Maybe this should inherit BaseWriteTask instead.

Reply:
yeah, that's my initial thought.  `BaseUpdateTask` is intended for tables that are stateful and need to be updated, so I think `BaseWriteTask` might be better.